src/sentinel.rs

```diff
@@ -218,8 +223,8 @@ impl<'a, Request, Name>
             .map(|(name, public_keys, _)| {
                 (name, public_keys.into_iter()
                                   .filter(|&(_, ref count)| *count >= self.keys_threshold)
-                                  .map(|(public_key, _)| public_key)
-                                  .collect::<Vec<PublicKey>>())})
+                                  .map(|(public_key, _)| PublicKey(public_key))
+                                  .collect::<_>())})
```

Comment:
I personally favour being explicit in collect::<_> statements; it might be redundant for the compiler, but it makes it very easy for a human to read what comes out of the chain of actions;